--- a/chaotic/CMakeLists.txt	2026-04-16 09:59:57.000000000 +0000
+++ b/chaotic/CMakeLists.txt	2026-05-08 21:02:33.592000000 +0000
@@ -2,6 +2,13 @@
 
 include(ChaoticGen)
 
+configure_file("${CMAKE_CURRENT_SOURCE_DIR}/bin/chaotic-gen" "${CMAKE_CURRENT_BINARY_DIR}/chaotic-gen" @ONLY)
+configure_file(
+    "${CMAKE_CURRENT_SOURCE_DIR}/bin-dynamic-configs/chaotic-gen-dynamic-configs"
+    "${CMAKE_CURRENT_BINARY_DIR}/chaotic-gen-dynamic-configs"
+    @ONLY
+)
+
 userver_module(
     chaotic
     SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}"
@@ -39,7 +46,12 @@
 )
 _userver_directory_install(
     COMPONENT chaotic
-    PROGRAMS "${CMAKE_CURRENT_SOURCE_DIR}/bin/chaotic-gen"
+    DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/bin-dynamic-configs"
+    DESTINATION "${CMAKE_INSTALL_LIBDIR}/userver"
+)
+_userver_directory_install(
+    COMPONENT chaotic
+    PROGRAMS "${CMAKE_CURRENT_BINARY_DIR}/chaotic-gen" "${CMAKE_CURRENT_BINARY_DIR}/chaotic-gen-dynamic-configs"
     DESTINATION "${CMAKE_INSTALL_BINDIR}"
 )
 _userver_directory_install(
--- a/chaotic/bin/chaotic-gen	2026-04-16 09:59:57.000000000 +0000
+++ b/chaotic/bin/chaotic-gen	2026-05-08 21:02:33.593000000 +0000
@@ -2,10 +2,11 @@
 
 BINDIR=`dirname "$0"`
 USERVER_PYTHON="${USERVER_PYTHON:-python3}"
+CHAOTIC_INSTALL_DIR="$BINDIR/../@CMAKE_INSTALL_LIBDIR@/userver"
 
 CHAOTIC_DIR="$BINDIR/.."
 if ! [ -e "$CHAOTIC_DIR/chaotic/main.py" ]; then
-    CHAOTIC_DIR="$BINDIR/../lib/userver/"
+    CHAOTIC_DIR="$CHAOTIC_INSTALL_DIR"
 fi
 
 export PYTHONPATH="$CHAOTIC_DIR"
--- a/chaotic/bin-dynamic-configs/chaotic-gen-dynamic-configs	2026-04-16 09:59:57.000000000 +0000
+++ b/chaotic/bin-dynamic-configs/chaotic-gen-dynamic-configs	2026-05-08 21:02:33.593000000 +0000
@@ -2,10 +2,11 @@
 
 BINDIR=`dirname "$0"`
 USERVER_PYTHON="${USERVER_PYTHON:-python3}"
+CHAOTIC_INSTALL_DIR="$BINDIR/../@CMAKE_INSTALL_LIBDIR@/userver"
 
 CHAOTIC_DIR="$BINDIR/.."
 if ! [ -e "$CHAOTIC_DIR/chaotic/main.py" ]; then
-    CHAOTIC_DIR="$BINDIR/../lib/userver/"
+    CHAOTIC_DIR="$CHAOTIC_INSTALL_DIR"
 fi
 
 export PYTHONPATH="$CHAOTIC_DIR"
--- a/chaotic-openapi/CMakeLists.txt	2026-04-16 09:59:57.000000000 +0000
+++ b/chaotic-openapi/CMakeLists.txt	2026-05-08 21:02:33.593000000 +0000
@@ -2,6 +2,8 @@
 
 include(ChaoticGen)
 
+configure_file("${CMAKE_CURRENT_SOURCE_DIR}/bin/chaotic-openapi-gen" "${CMAKE_CURRENT_BINARY_DIR}/chaotic-openapi-gen" @ONLY)
+
 userver_module(
     chaotic-openapi
     SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}"
@@ -36,3 +38,8 @@
     DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/chaotic_openapi"
     DESTINATION "${CMAKE_INSTALL_LIBDIR}/userver"
 )
+_userver_directory_install(
+    COMPONENT chaotic-openapi
+    PROGRAMS "${CMAKE_CURRENT_BINARY_DIR}/chaotic-openapi-gen"
+    DESTINATION "${CMAKE_INSTALL_BINDIR}"
+)
--- a/chaotic-openapi/bin/chaotic-openapi-gen	2026-04-16 09:59:57.000000000 +0000
+++ b/chaotic-openapi/bin/chaotic-openapi-gen	2026-05-08 21:02:33.593000000 +0000
@@ -2,11 +2,14 @@
 
 BINDIR=`dirname "$0"`
 USERVER_PYTHON="${USERVER_PYTHON:-python3}"
+CHAOTIC_INSTALL_DIR="$BINDIR/../@CMAKE_INSTALL_LIBDIR@/userver"
 
 CHAOTIC_DIR="$BINDIR/..:$BINDIR/../../chaotic"
-if ! [ -e "$BINDIR/../chaotic_openapi/main.py" ]; then
-    CHAOTIC_DIR="$BINDIR/../lib/userver/"
+CHAOTIC_MAIN="$BINDIR/../chaotic_openapi/main.py"
+if ! [ -e "$CHAOTIC_MAIN" ]; then
+    CHAOTIC_DIR="$CHAOTIC_INSTALL_DIR"
+    CHAOTIC_MAIN="$CHAOTIC_DIR/chaotic_openapi/main.py"
 fi
 
 export PYTHONPATH="$CHAOTIC_DIR"
-exec "$USERVER_PYTHON" "$BINDIR/../chaotic_openapi/main.py" "$@"
+exec "$USERVER_PYTHON" "$CHAOTIC_MAIN" "$@"
